cmd/ci-operator/main.go

```diff
@@ -121,14 +116,34 @@ func (o *options) Run() error {
 			return fmt.Errorf("could not get image client for cluster config: %v", err)
 		}
 
-		if _, err := imageGetter.ImageStreams(o.jobSpec.Identifier()).Create(&imageapi.ImageStream{
+		// create the image stream or read it to get its uid
+		is, err := imageGetter.ImageStreams(o.jobSpec.Namespace()).Create(&imageapi.ImageStream{
 			ObjectMeta: meta.ObjectMeta{
-				Namespace: o.jobSpec.Identifier(),
+				Namespace: o.jobSpec.Namespace(),
 				Name:      steps.PipelineImageStream,
 			},
-		}); err != nil && ! errors.IsAlreadyExists(err) {
-			return fmt.Errorf("could not set up pipeline imagestream for test: %v", err)
+		})
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not set up pipeline imagestream for test: %v", err)
+			}
+			is, _ = imageGetter.ImageStreams(o.jobSpec.Namespace()).Get(steps.PipelineImageStream, meta.GetOptions{})
 		}
+		if is != nil {
+			isTrue := true
+			o.jobSpec.SetOwner(&meta.OwnerReference{
```

Comment:
Aren't we namespacing everything anyway? Where does this help us get more granular ownership?

Reply:
If you delete the pipeline everything goes away, which makes it easier to test and iterate in a namespace.  More of a best practice than a hard requirement.